### ISSUE TITLE
Cannot train pytorch sar_resnet31 and master recognition model

### ISSUE THREAD

**Issue by alexcailletMonk** [mindee/doctr#802]
### Bug description

We tried to train sar_resnet31 and master with `references/recognition/train_pytorch.py`. In both cases, it doesn't train. Train loss, train val are at 0 and exact and partial match too.
Training works with VGG.

### Code snippet to reproduce the bug

```python train_pytorch.py train_data path val_data_path sar_resnet31 --wb -b 64```
```python train_pytorch.py train_data path val_data_path master --wb -b 64```

### Error traceback

no error

### Environment

DocTR version: 0.5.1a0
TensorFlow version: 2.6.2
PyTorch version: 1.9.0+cu111 (torchvision 0.10.0+cu111)
OpenCV version: 4.5.4-dev
OS: Debian GNU/Linux 10 (buster)
Python version: 3.8.12
Is CUDA available (TensorFlow): Yes
Is CUDA available (PyTorch): Yes
CUDA runtime version: 11.4.120
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 470.57.02
cuDNN version: Could not collect

- [x] #920 
- [x] #925
- [x] #941 
- [x] #949

**Comment by fg-mindee** [comment]
Thanks @alexcailletMonk!

I was just able to reproduce the error, I'll investigate! We haven't spent much time on SAR & MASTER lately because of their latency compared to other architectures, but we want to catch up with the others :ok_hand:

**Comment by fg-mindee** [comment]
Hi @alexcailletMonk :wave: 

Quick update on this with ongoing & incoming PRs:
- [ ] #831 
- [ ] Fixing MASTER implementation in PyTorch

**Comment by felixdittrich92** [comment]
@frgfm @charlesmindee After some debugging i have MASTER fixed (with some look on https://github.com/wenwenyu/MASTER-pytorch) so i will cleanup a bit and open a PR that we can have a look at improvements :+1:

**Comment by felixdittrich92** [comment]
- [ ] fix MASTER implementation (PyTorch and Tensorflow)

- [ ] fix SAR implementation (PyTorch and Tensorflow)

- [ ] add pretrained checkpoints for both models in both frameworks

**Comment by felixdittrich92** [comment]
- [ ] fix MASTER implementation (PyTorch and Tensorflow)

- [ ] fix SAR implementation (PyTorch and Tensorflow)

- [ ] add pretrained checkpoints for both models in both frameworks

**Comment by felixdittrich92** [comment]
short update:
I have some internal work to do first, once I'm done with that I'll move on to the MASTER implementation :)
Maybe we could collaborate on this @frgfm ?

**Comment by frgfm** [comment]
Happy to do so, but I'm still waiting on my new setup (in the meantime, I have limited capacity for GPU-intensive tasks and docker) 😅

**Comment by felixdittrich92** [comment]
@frgfm yes no stress :)

**Comment by frgfm** [comment]
Yes, adding new checkpoints will be for 0.6.0 !
But fixing the implementation (being able to correctly train and performance inference with them in both framework) is the topic of this issue :+1:

**Comment by felixdittrich92** [comment]
Closed by: #920 #925 #941 #949
